ddtrace/appsec/iast/taint_sinks/_base.py

```diff
@@ -49,14 +49,12 @@ def report(cls, evidence_value=""):
 
         TODO: check deduplications if DD_IAST_DEDUPLICATION_ENABLED is true
         """
-        if cls.has_quota():
-
+        if cls.acquire_quota():
             span = tracer.current_root_span()
             if not span:
                 log.debug("No root span in the current execution. Skipping IAST taint sink.")
                 return None
```

Comment:
I'd move cls.aquire_quota to after this check, to avoid aquiring_quota for nothing